#Python Data Cleaning




Input Dataset

**[nls97.csv](https://drive.google.com/file/d/1-av28HdgEAYPc7ZSmvIh1JdR6Pb4HIfg/view?usp=sharing)** - the National Longitudinal Survey (NLS) 1997-2017.


1. ทำการ Cleaning ข้อมูล nls97.csv ให้เปลี่ยนค่า NaN เป็นค่า 0 ทั้งชุดข้อมูล
2. ทำการ Filter และแสดงข้อมูล row index ที่ 100:800 เฉพาะ column ชื่อ gender	birthmonth	birthyear	highestgradecompleted wageincome	weeklyhrscomputer	weeklyhrstv โดยใช้คำสั่ง .loc[ ] และ .iloc[ ]
3. ทำการ Filter และแสดงข้อมูลเฉพาะคนที่เกิดตั้งแต่ 1980 - 1983
4. แสดงผล Summary Statistics ของ nls97 และเขียนวิเคราะห์ข้อมูลทางสถิติที่ได้ของกลุ่มประชากรชุดนี้

## 1. โปรแกรม Cleaning ข้อมูล nls97.csv ให้เปลี่ยนค่า NaN เป็นค่า 0 ทั้งชุดข้อมูล

In [ ]:
nls97.fillna(0, inplace=True)
print(nls97.isnull().sum())
nls97

gender                   0
birthmonth               0
birthyear                0
highestgradecompleted    0
maritalstatus            0
                        ..
colenroct15              0
colenrfeb16              0
colenroct16              0
colenrfeb17              0
colenroct17              0
Length: 88, dtype: int64


gender  birthmonth  birthyear  highestgradecompleted  maritalstatus  \
personid                                                                        
100061    Female           5       1980                   13.0        Married   
100139      Male           9       1983                   12.0        Married   
100284      Male          11       1984                    7.0  Never-married   
100292      Male           4       1982                    0.0              0   
100583      Male           1       1980                   13.0        Married   
...          ...         ...        ...                    ...            ...   
968973      Male          11       1980                   12.0        Married   
968984      Male           3       1981                   12.0  Never-married   
969226    Female           6       1980                   12.0  Never-married   
969334    Female           1       1983                    9.0  Never-married   
969336      Male           9       1980                    0.0              0   

          childathome  childnotathome  wageincome        weeklyhrscomputer  \
personid                                                                     
100061            4.0             0.0     12500.0  10 hours or more a week   
100139            2.0             0.0    120000.0      1 to 3 hours a week   
100284            1.0             0.0     58000.0                     None   
100292            0.0             0.0         0.0                        0   
100583            4.0             0.0     30000.0  Less than 1 hour a week   
...               ...             ...         ...                      ...   
968973            1.0             2.0     80000.0  10 hours or more a week   
968984            1.0             0.0    120000.0  10 hours or more a week   
969226            4.0             0.0     17000.0                     None   
969334            6.0             0.0      5000.0                     None   
969336            0.0             0.0         0.0                        0   

                         weeklyhrstv  ...      colenrfeb13      colenroct13  \
personid                              ...                                     
100061         11 to 20 hours a week  ...  1. Not enrolled  1. Not enrolled   
100139          3 to 10 hours a week  ...  1. Not enrolled  1. Not enrolled   
100284         11 to 20 hours a week  ...  1. Not enrolled  1. Not enrolled   
100292                             0  ...  1. Not enrolled  1. Not enrolled   
100583          3 to 10 hours a week  ...  1. Not enrolled  1. Not enrolled   
...                              ...  ...              ...              ...   
968973         11 to 20 hours a week  ...  1. Not enrolled  1. Not enrolled   
968984          3 to 10 hours a week  ...  1. Not enrolled  1. Not enrolled   
969226    Less than 2 hours per week  ...  1. Not enrolled  1. Not enrolled   
969334    Less than 2 hours per week  ...  1. Not enrolled  1. Not enrolled   
969336                             0  ...                0                0   

              colenrfeb14      colenroct14      colenrfeb15      colenroct15  \
personid                                                                       
100061    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
100139    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
100284    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
100292    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
100583    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
...                   ...              ...              ...              ...   
968973    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
968984    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
969226    1. Not enrolled  1. Not enrolled  1. Not enrolled  1. Not enrolled   
969334    1. Not enrolled  1. Not

##2. ทำการ Filter และแสดงข้อมูล row index ที่ 100:800 เฉพาะ column ชื่อ gender birthmonth birthyear highestgradecompleted wageincome weeklyhrscomputer weeklyhrstv โดยใช้คำสั่ง .loc[ ] และ .iloc[ ]

In [ ]:
analysisdemofilter = nls97.iloc[100:800, [0,1,2,3,4,8,9,10]]
print(type(analysisdemofilter))
print(analysisdemofilter)

<class 'pandas.core.frame.DataFrame'>
     personid  gender  birthmonth  birthyear  highestgradecompleted  \
100    109234    Male          11       1983                   12.0   
101    109255    Male           6       1984                   16.0   
102    109282  Female           7       1984                    NaN   
103    109306    Male          11       1981                    NaN   
104    109316  Female           9       1982                    8.0   
..        ...     ...         ...        ...                    ...   
795    177947    Male           1       1983                   12.0   
796    177987    Male          10       1982                   20.0   
797    178022  Female           4       1984                   14.0   
798    178083  Female           9       1982                   20.0   
799    178137    Male           7       1980                   12.0   

     wageincome        weeklyhrscomputer                 weeklyhrstv  
100         NaN                     No

##3. ทำการ Filter และแสดงข้อมูลเฉพาะคนที่เกิดตั้งแต่ 1980 - 1983

In [ ]:
Agefilter = analysisdemo['birthyear'] <=1983
analysisdemo[Agefilter]

gender  birthmonth  birthyear  highestgradecompleted  wageincome  \
100    Male          11       1983                   12.0         NaN   
103    Male          11       1981                    NaN         NaN   
104  Female           9       1982                    8.0         NaN   
105    Male           3       1983                    NaN         NaN   
106    Male          10       1981                   11.0      5000.0   
..      ...         ...        ...                    ...         ...   
795    Male           1       1983                   12.0     30000.0   
796    Male          10       1982                   20.0      1900.0   
798  Female           9       1982                   20.0     70000.0   
799    Male           7       1980                   12.0         NaN   
800  Female          11       1983                    NaN         NaN   

           weeklyhrscomputer                 weeklyhrstv  
100                     None        3 to 10 hours a week  
103                      NaN                         NaN  
104                     None       11 to 20 hours a week  
105                      NaN                         NaN  
106  10 hours or more a week   More than 40 hours a week  
..                       ...                         ...  
795  10 hours or more a week        3 to 10 hours a week  
796  10 hours or more a week  Less than 2 hours per week  
798  10 hours or more a week        3 to 10 hours a week  
799  10 hours or more a week        3 to 10 hours a week  
800                      NaN                         NaN  

[564 rows x 7 columns]

##4. แสดงผล Summary Statistics ของ nls97 และเขียนวิเคราะห์ข้อมูลทางสถิติที่ได้ของกลุ่มประชากรชุดนี้

In [ ]:
nls97.describe()

birthmonth    birthyear  highestgradecompleted  childathome  \
count  8697.000000  8697.000000            8697.000000  8697.000000   
mean      6.561573  1982.006784              10.418995     0.991376   
std       3.466214     1.398374               7.239569     1.306094   
min       1.000000  1980.000000               0.000000     0.000000   
25%       3.000000  1981.000000               0.000000     0.000000   
50%       7.000000  1982.000000              12.000000     0.000000   
75%      10.000000  1983.000000              16.000000     2.000000   
max      12.000000  1984.000000              95.000000     9.000000   

       childnotathome     wageincome  nightlyhrssleep    satverbal  \
count     8697.000000    8697.000000      8697.000000  8697.000000   
mean         0.218581   28086.600092         4.950328    78.381281   
std          0.721075   39258.667061         3.118197   187.117344   
min          0.000000       0.000000         0.000000     0.000000   
25%          0.000000       0.000000         0.000000     0.000000   
50%          0.000000   15000.000000         6.000000     0.000000   
75%          0.000000   45000.000000         7.000000     0.000000   
max         12.000000  235884.000000        20.000000   800.000000   

           satmath   gpaoverall  ...  weeksworked08  weeksworked09  \
count  8697.000000  8697.000000  ...    8697.000000    8697.000000   
mean     78.544441   188.248247  ...      35.810279      33.878694   
std     187.636200   141.980279  ...      21.325868      22.226879   
min       0.000000     0.000000  ...       0.000000       0.000000   
25%       0.000000     0.000000  ...      16.000000       5.000000   
50%       0.000000   243.000000  ...      51.000000      50.000000   
75%       0.000000   306.000000  ...      52.000000      52.000000   
max     800.000000   417.000000  ...      52.000000      52.000000   

       weeksworked10  weeksworked11  weeksworked12  weeksworked13  \
count    8697.000000    8697.000000    8697.000000    8697.000000   
mean       33.166149      33.632402      32.793032      33.140278   
std        22.737651      23.172382      23.583669      23.384488   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%        50.000000      50.000000      52.000000      52.000000   
75%        52.000000      53.000000      52.000000      52.000000   
max        52.000000      53.000000      52.000000      52.000000   

       weeksworked14  weeksworked15  weeksworked16  weeksworked17  
count    8697.000000    8697.000000    8697.000000    8697.000000  
mean       32.463378      32.556284      30.792917      28.980223  
std        23.756383      23.592408      24.789649      23.910387  
min         0.000000       0.000000       0.000000       0.000000  
25%         0.000000       0.000000       0.000000       0.000000  
50%        52.000000      51.000000      51.000000      44.000000  
75%        52.000000      52.000000      53.000000      52.000000  
max        52.000000      52.000000      53.000000      52.000000  

[8 rows x 31 columns]

กลุ่มประชากรในสถิติมี 8697 คน childnotathome ค่าน้อยสุดมันคือ 0 ค่ามากสุดมันคือ 12